### PR TITLE
Rename `aws_types::config::Config` to `sdk_config::SdkConfig`

### PR DESCRIPTION
**Pull request by Velfi**
## Motivation and Context

aws-sdk-rust#406

## Description

rename `aws_types::config::Config` to `sdk_config::SdkConfig` and update everything referencing the old name

## Testing

ran existing tests

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/rust-runtime/aws-types/src/lib.rs [hunk 1]
```diff
@@ -15,11 +15,11 @@
 
 pub mod app_name;
 pub mod build_metadata;
-pub mod config;
 pub mod credentials;
 #[doc(hidden)]
 pub mod os_shim_internal;
 pub mod region;
+pub mod sdk_config;
```
Comment (rcoh) [inline aws/rust-runtime/aws-types/src/lib.rs:22]: do you think we should just re-export SdkConfig into the root?

Comment (rcoh) [reply aws/rust-runtime/aws-types/src/lib.rs:22]: the duplication required by `sdk_config::SdkConfig` seems a little bit cumersome

**Review by rcoh — COMMENTED** [review]
lgtm, let's just discuss possible re-export

File: aws/rust-runtime/aws-config/src/lib.rs [hunk 2]
```diff
@@ -108,14 +107,15 @@ pub fn from_env() -> ConfigLoader {
...
-pub async fn load_from_env() -> aws_types::config::Config {
+pub async fn load_from_env() -> aws_types::SdkConfig {
     from_env().load().await
 }
 
 /// Load default sources for all configuration with override support
 pub use loader::ConfigLoader;
 
 mod loader {
+    use hyper::client::HttpConnector;
```
Comment (Velfi) [inline aws/rust-runtime/aws-config/src/lib.rs:118]:
```suggestion
```

File: CHANGELOG.next.toml [hunk 3]
```diff
@@ -17,6 +17,76 @@ references = ["smithy-rs#1225"]
 meta = { "breaking" = false, "tada" = false, "bug" = false }
 author = "Velfi"
 
+[[aws-sdk-rust]]
+message = """
+`aws_types::config::Config` has been renamed to `aws_types::sdk_config::SdkConfig`. This is to better differentiate it
+from service-specific configs like `aws_s3_sdk::Config`. If you were creating shared configs with
```
Comment (rcoh) [inline CHANGELOG.next.toml:23]:
```suggestion
from service-specific configs like `aws_sdk_s3::Config`. If you were creating shared configs with
```

File: CHANGELOG.next.toml [hunk 4]
```diff
@@ -17,6 +17,76 @@ references = ["smithy-rs#1225"]
...
+    let config = Config::builder()
+    // config builder methods...
+    .build()
+    .await;
+}
+```
+
+_After:_
+```rust
+use aws_types::sdk_config::SdkConfig;
```
Comment (rcoh) [inline CHANGELOG.next.toml:76]: can you update this for the re-export?

**Review by rcoh — APPROVED** [review]
LGTM!

**Review by rcoh — APPROVED** [review]
LGTM, could consider also deprecating the config module.

**Pull request merged**